websockify/websocketproxy.py

```diff
@@ -56,7 +56,8 @@ def send_auth_error(self, ex):
         self.end_headers()
     
     def validate_connection(self):
-        if self.server.token_plugin:
+        # don't use token for non-ws requests
+        if self.server.token_plugin and self.headers.get('upgrade') and self.headers.get('upgrade').lower() == 'websocket':
```

Comment:
This is just messy. I think it's better to split things up instead. Create a new `auth_connection()` or something like that.

Reply:
Agreed it was messy, but this is not how it ended up after the latter commits.

Reply:
You still had to add the exception via the extra parameter. So I still say split things up for clarity.

Reply:
Alright, I'll take care of that later today.

Reply:
Done.